src/react/DevicesPage.tsx

```diff
@@ -50,7 +66,111 @@ const matchMetadata = (key, value, filterKey, filterValue) => {
   return true;
 };
 
-const matchFilters = (device, filters) => {
+interface MatchedMetadataProps {
+  filters: DeviceFilters;
+  metadata: AstarteDevice['metadata'];
+}
+
+const MatchedMetadata = ({
+  filters,
+  metadata,
+}: MatchedMetadataProps): React.ReactElement | null => {
+  const { metadataKey = '', metadataValue = '' } = filters;
+
+  if (metadataKey === '' && metadataValue === '') {
+    return null;
+  }
+
+  return (
+    <>
+      {Array.from(metadata)
+        .filter(([key, value]) => matchMetadata(key, value, metadataKey, metadataValue))
+        .map(([key, value]) => (
+          <div key={key} style={{ overflowWrap: 'anywhere' }}>
+            <Highlight word={metadataKey}>{key}</Highlight>
+            {': '}
+            <Highlight word={metadataValue}>{value}</Highlight>
+          </div>
+        ))}
+    </>
+  );
+};
+
+const CircleIcon = React.forwardRef<HTMLElement, React.HTMLProps<HTMLElement>>(
+  ({ children, className, ...props }, ref) => (
+    <i ref={ref} {...props} className={`fas fa-circle ${className}`}>
+      {children}
+    </i>
+  ),
+);
+
+interface DeviceRowProps {
+  device: AstarteDevice;
+  filters: DeviceFilters;
+}
+
+const DeviceRow = ({ device, filters }: DeviceRowProps): React.ReactElement => {
+  let colorClass;
+  let lastEvent;
+  let tooltipText;
+
+  if (device.isConnected) {
+    tooltipText = 'Connected';
+    colorClass = 'icon-connected';
+    lastEvent = `Connected on ${(device.lastConnection as Date).toLocaleString()}`;
+  } else if (device.lastConnection) {
+    tooltipText = 'Disconnected';
+    colorClass = 'icon-disconnected';
+    lastEvent = `Disconnected on ${(device.lastDisconnection as Date).toLocaleString()}`;
+  } else {
+    tooltipText = 'Never connected';
+    colorClass = 'icon-never-connected';
+    lastEvent = 'Never connected';
+  }
+
+  return (
+    <tr>
+      <td>
+        <OverlayTrigger
+          placement="right"
+          delay={{ show: 150, hide: 400 }}
+          overlay={<Tooltip id={device.id}>{tooltipText}</Tooltip>}
```

Comment:
you forgot to copy the element style. You could also move that to a specific css class, but right now it's missing from the code

Reply:
I did not forget, I stripped it because it wasn't sorting any effect since OverlayTrigger doesn't expose a `style` property.